Mathlib/Lean/Data/NameMap.lean

```diff
@@ -0,0 +1,24 @@
+/-
+Copyright (c) 2023 Jon Eugster. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Eugster
+-/
+import Lean.Data.NameMap
+
+/-!
+# Additional functions on `Lean.NameMap`.
+
+We provide `NameMap.filter`.
+-/
+
+namespace Lean.NameMap
+
+/--
+Filter a `NameMap`. Takes a second optional argument `modify : α → α` which allows
+to modify/filter the value of each (key, value)-pair simultaneously.
+-/
+def filter (m : NameMap α) (f : Name → Bool) (modify : α → α := id) : NameMap α :=
+  m.fold process {}
```

Comment:
For consistency with other APIs, could we have `filter (f : Name → α → Bool)`, and then `filterMap (f : Name → α → Option α)`?

Reply:
Done that. I only need `NameMap.filterMap` and not `NameMap.filter`, but in praxis I would more likely use `filter`
with a function `(f : Name → Bool)`:

```lean
graph := graph.filter (isPrefixOf p)
-- vs
graph := graph.filter (fun n _ => isPrefixOf p n)
```
I named the two versions `filterName` and `filter`, but maybe there is a better name.